packages/mobile/src/language/Language.tsx

```diff
@@ -12,25 +12,31 @@ import { CustomEventNames } from 'src/analytics/constants'
 import { setLanguage } from 'src/app/actions'
 import { AVAILABLE_LANGUAGES } from 'src/config'
 import { Namespaces } from 'src/i18n'
+import { emptyHeader, headerWithBackButton } from 'src/navigator/Headers.v2'
 import { navigate, navigateBack } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 
-type Props = StackScreenProps<StackParamList, Screens.Language>
+type RouterProps = StackScreenProps<StackParamList, Screens.Language>
+type Props = RouterProps
 
 export default function Language({ route }: Props) {
   const dispatch = useDispatch()
   const { t, i18n } = useTranslation(Namespaces.accountScreen10)
+  const fromSettings = route.params?.fromSettings
 
   function onSelect(language: string, code: string) {
     CeloAnalytics.track(CustomEventNames.language_select, { language, selectedAnswer: code })
-    const nextScreen = route.params?.nextScreen ?? Screens.JoinCelo
     dispatch(setLanguage(code))
-    if (nextScreen === 'GO_BACK') {
-      navigateBack()
-    } else {
-      navigate(nextScreen)
-    }
+    // Wait for next frame before navigating back
+    // so the user can see the changed selection briefly
+    requestAnimationFrame(() => {
```

Comment:
i think this technically goes on the next tick, so i wouldn't be surprised if in the prod build it's not visible because it happens so fast. i would recommend setting an explict ms timeout if you want to be sure it's seen

Reply:
I would say, that using constant timeout on different hardware would be inconstant as well. Some phones are smooth and fast and some are not. So on a faster phones it could be to slow and on the old ones still would not be enough. I piggy back this approach from Currency screen, where @jeanregisser has implemented this way.